File: Dependant variables_webComponents/src/app/services/NorthWindv2API-service.ts
import { CustomerDto } from '../models/north-windv-2api/customer-dto';
import { OrderDetailDto } from '../models/north-windv-2api/order-detail-dto';
import { OrderDto } from '../models/north-windv-2api/order-dto';
import { BehaviorSubject } from 'rxjs';

const API_ENDPOINT = 'https://data-northwind.indigo.design';

class NorthWindv2APIService {

	public selectedOrder: BehaviorSubject<OrderDto | undefined> = new BehaviorSubject<OrderDto | undefined>(undefined);

	private _selectedCustomer$!: BehaviorSubject<CustomerDto | undefined>;
  
	public get selectedCustomer(): BehaviorSubject<CustomerDto | undefined> {
		if (!this._selectedCustomer$) {
			this._selectedCustomer$ = new BehaviorSubject<CustomerDto | undefined>(undefined);
			this._selectedCustomer$.subscribe(() => this.selectedOrder.next(undefined));
		}
  
		return this._selectedCustomer$;
	}

	public getCustomerDtoList = async (): Promise<CustomerDto[]> => {
		const response = await fetch(`${API_ENDPOINT}/Customers`);
		if (!response.ok) {
			return Promise.reject(response.statusText);
		}
		return response.json();
	}

	public getOrderDetailDtoList = async (id: string): Promise<OrderDetailDto[]> => {
		const response = await fetch(`${API_ENDPOINT}/Orders/${id}/Details`);
		if (!response.ok) {
			return Promise.reject(response.statusText);
		}
		return response.json();
	}

	public getOrderDtoList = async (id: string): Promise<OrderDto[]> => {
		const response = await fetch(`${API_ENDPOINT}/Customers/${id}/Orders`);
		if (!response.ok) {
			return Promise.reject(response.statusText);
		}
		return response.json();
	}
}
export const northWindv2APIService: NorthWindv2APIService = new NorthWindv2APIService();

File: Dependant variables_blazor-wasm/DependantVariables/Services/NorthWindv2APIService.cs
using System.Net.Http.Json;
using System.Reactive.Subjects;
using DependantVariables.Models.NorthWindv2API;

namespace DependantVariables.NorthWindv2API
{
    public class NorthWindv2APIService: INorthWindv2APIService
    {
        private readonly HttpClient _http;

        public NorthWindv2APIService(HttpClient http)
        {
            _http = http;
        }
        public BehaviorSubject<OrderDto> SelectedOrder { get; set; } = new(default);

        private BehaviorSubject<CustomerDto> _selectedCustomer;
        public BehaviorSubject<CustomerDto> SelectedCustomer
        {
            get
            {
                // Selected order depends on selected customer
                if (_selectedCustomer == null)
                {
                    _selectedCustomer = new(default);
                    SelectedOrder = new(default);
                    _selectedCustomer.Subscribe(async _ => SelectedOrder.OnNext(default));
                }
                
                return _selectedCustomer;
            }
        }

        public async Task<List<CustomerDto>> GetCustomerDtoList()
        {
            using HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, new Uri("https://data-northwind.indigo.design/Customers", UriKind.RelativeOrAbsolute));
            using HttpResponseMessage response = await _http.SendAsync(request).ConfigureAwait(false);
            if (response.IsSuccessStatusCode)
            {
                return await response.Content.ReadFromJsonAsync<List<CustomerDto>>().ConfigureAwait(false);
            }

            return new List<CustomerDto>();
        }

        public async Task<List<OrderDetailDto>> GetOrderDetailDtoList(string id)
        {
            using HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, new Uri($"https://data-northwind.indigo.design/Orders/{id}/Details", UriKind.RelativeOrAbsolute));
            using HttpResponseMessage response = await _http.SendAsync(request).ConfigureAwait(false);
            if (response.IsSuccessStatusCode)
            {
                return await response.Content.ReadFromJsonAsync<List<OrderDetailDto>>().ConfigureAwait(false);
            }

            return new List<OrderDetailDto>();
        }

        public async Task<List<OrderDto>> GetOrderDtoList(string id)
        {
            using HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, new Uri($"https://data-northwind.indigo.design/Customers/{id}/Orders", UriKind.RelativeOrAbsolute));
            using HttpResponseMessage response = await _http.SendAsync(request).ConfigureAwait(false);
            if (response.IsSuccessStatusCode)
            {
                return await response.Content.ReadFromJsonAsync<List<OrderDto>>().ConfigureAwait(false);
            }

            return new List<OrderDto>();
        }
    }
}
